src/matrixeventprocessor.ts

```diff
@@ -137,6 +137,10 @@ export class MatrixEventProcessor {
             return;
         } else if (event.type === "m.room.message" || event.type === "m.sticker") {
             log.verbose(`Got ${event.type} event`);
+            if (this.config.bridge.userBlacklist.includes(event.sender)) {
```

Comment:
shouldn't this blacklist be applied further up, to also prevent messages on `m.room.member` and thelike? When doing that, what if that user redacts someone elses message, should that redaction be bridged?

Reply:
I think we would want to drop any events _sent by_ this user, if we're implementing a true blacklist.

Reply:
Then we would also need to drop the avatar + username changes in `m.room.member`, though

Reply:
I was worried that ignoring things like invite events would corrupt the state of the bridge's view of a room somehow. Thinking about it some more, though, if all state events pertaining to a blacklisted user are ignored, there should be no problem, as that user will never have a chance to affect the state of the room.

Ignoring redactions is a good idea too.

How about ignoring the following:

* All state events related to a blacklisted user (state_key = user)
* All events sent by a blacklisted user (sender = user)

Reply:
soru means ignoring the "mxid has joined the room on matrix" messages, not necessarily ingorning the invite internally

Reply:
this method [here](https://github.com/Half-Shot/matrix-appservice-discord/blob/master/src/matrixeventprocessor.ts#L187)